src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1051,6 +1061,52 @@ private void RegisterHomeWorkspace(HomeWorkspaceModel newWorkspace)
 
         #endregion
 
+        #region backup/timer
+
+        /// <summary>
+        /// Backup all the files
+        /// </summary>
+        protected void SaveBackupFiles(object state)
+        {
+            foreach (var workspace in Workspaces)
+            {
+                if (!workspace.HasUnsavedChanges)
+                    continue;
+
+                string fileName;
+                if (string.IsNullOrEmpty(workspace.FileName))
+                {
+                    fileName = "recent_new_file_" + workspace.Guid;
+                    var ext = workspace is HomeWorkspaceModel ? ".DYN" : ".DYF";
+                    fileName += ext;
+                }
+                else
+                {
+                    fileName = Path.GetFileName(workspace.FileName);
+                }
+
+                var savePath = Path.Combine(pathManager.BackupDirectory, fileName);
+                workspace.SaveAs(savePath, null);
+                Logger.Log("Backup file is saved: " + savePath);
+            }
+        }
+
+        /// <summary>
+        /// Start the timer to backup files periodically
+        /// </summary>
+        public void StartBackupFilesTimer()
```

Comment:
This method needs to be made `private`, and then if `backupFilesTimer != null`, we should throw an exception rather than silently changing the backup timing. Does that make sense?